libraries/rush-lib/src/logic/operations/ShardedPhaseOperationPlugin.ts

```diff
@@ -0,0 +1,212 @@
+
+      const collatorDisplayName: string = `${getDisplayName(phase, project)} - collate`;
+      const collatorRawScript: string | undefined = getScriptToRun(project, phase.name, undefined);
+
+      if (collatorRawScript === undefined && phase.missingScriptBehavior === 'error') {
+        throw new Error(
+          `The project '${project.packageName}' does not define a '${phase.name}' command in the 'scripts' section of its package.json`
+        );
+      }
+
+      if (collatorRawScript) {
+        const collatorRunner: ShellOperationRunner = new ShellOperationRunner({
+          commandToRun: collatorRawScript,
+          displayName: collatorDisplayName,
+          phase,
+          rushConfiguration,
+          rushProject: project,
+          operationSettings,
+          environment: {
+            ...process.env,
+            [EnvironmentVariableNames.RUSH_SHARD_PARENT_FOLDER]: parentFolder,
+            [EnvironmentVariableNames.RUSH_SHARD_COUNT]: shards.toString()
+          }
+        });
+        collatorNode.runner = collatorRunner;
+      } else {
+        collatorNode.runner = new NullOperationRunner({
+          name: collatorDisplayName,
+          result: OperationStatus.NoOp,
+          silent: phase.missingScriptBehavior === 'silent'
+        });
+      }
+      for (const dependent of operation.consumers) {
+        dependent.addDependency(collatorNode);
+        dependent.deleteDependency(operation);
+      }
+
+      const customParameters: readonly string[] = getCustomParameterValuesForPhase(phase);
+      const baseCommand: string | undefined = getScriptToRun(
+        project,
+        `${phase.name}:shard`,
+        phase.shellCommand
+      );
+
+      const collatorMissingScriptBehavior: string =
+        shardScriptConfiguration?.missingScriptBehavior ?? 'error';
+      if (baseCommand === undefined && collatorMissingScriptBehavior === 'error') {
+        throw new Error(
+          `The project '${project.packageName}' does not define a '${phase.name}:shard' command in the 'scripts' section of its package.json`
+        );
+      }
+
+      for (let shard: number = 1; shard <= shards; shard++) {
+        const shardOperation: Operation = new Operation({
+          project,
+          phase
+        });
+
+        // Add the shard argument to the custom parameters, replacing any templated values.
+        const shardArgumentFormat: string =
+          operationSettings.sharding.shardArgumentFormat ??
+          `--shard=${TemplateStrings.SHARD_INDEX}/${TemplateStrings.SHARD_COUNT}`;
+
+        const shardArgument: string = shardArgumentFormat
+          .replace(TemplateStrings.SHARD_INDEX, shard.toString())
+          .replace(TemplateStrings.SHARD_COUNT, shards.toString());
+
+        const outputFolderArgumentFlag: string =
+          operationSettings.sharding.outputFolderArgument?.argumentFlag ?? '--shard-output-directory';
+        const outputDirectory: string = path.join(parentFolder, shard.toString());
```

Comment:
```suggestion
        const outputDirectory: string = `${parentFolder}/${shard}`;
```
This preserves use of `/` as the directory separator, which prevents the cache key from changing between Windows and Linux.